tests/test_web/test_api_v1.py

```diff
@@ -1924,3 +1937,96 @@ def test_create_empty_index_not_allowed_params(mock_smfsc, data, error_msg, clie
     assert rv.status_code == 400
     assert error_msg == rv.json['error']
     mock_smfsc.assert_not_called()
+
+
+@pytest.mark.parametrize(
+    ('related_bundles_content', 'expired', 'finalized', 'expected'),
+    (
+        (
+            ['foobar'],
+            False,
+            False,
+            {'status': 400, 'mimetype': 'application/json', 'json': {'error': mock.ANY}},
+        ),
+        (
+            ['foobar'],
+            True,
+            False,
+            {'status': 400, 'mimetype': 'application/json', 'json': {'error': mock.ANY}},
+        ),
+        (
+            ['foobar'],
+            False,
+            True,
+            {'status': 200, 'mimetype': 'application/json', 'json': ['foobar']},
+        ),
+        (
+            [],
+            True,
+            False,
+            {'status': 400, 'mimetype': 'application/json', 'json': {'error': mock.ANY}},
+        ),
+        (
+            None,
+            False,
+            False,
+            {'status': 400, 'mimetype': 'application/json', 'json': {'error': mock.ANY}},
+        ),
+        (
+            None,
+            False,
+            True,
+            {'status': 404, 'mimetype': 'application/json', 'json': {'error': mock.ANY}},
+        ),
+        (
+            None,
+            True,
+            True,
+            {'status': 410, 'mimetype': 'application/json', 'json': {'error': mock.ANY}},
+        ),
+    ),
+)
+def test_get_build_related_bundles(
+    client,
+    db,
+    minimal_request_regenerate_bundle,
+    tmpdir,
+    related_bundles_content,
+    expired,
+    finalized,
+    expected,
+):
+    minimal_request_regenerate_bundle.add_state('in_progress', 'Starting things up!')
+    db.session.commit()
+
+    client.application.config['IIB_REQUEST_RELATED_BUNDLES_DIR'] = str(tmpdir)
+    if expired:
+        client.application.config['IIB_REQUEST_DATA_DAYS_TO_LIVE'] = -1
+    if finalized:
+        minimal_request_regenerate_bundle.add_state('complete', 'The request is complete')
+        db.session.commit()
+    request_id = minimal_request_regenerate_bundle.id
+    related_bundles_file = tmpdir.join(f'{request_id}_related_bundles.json')
+    if related_bundles_content is not None:
```

Comment:
Could we have a new test? I don't see a reason to use if statements in tests.

Reply:
Can I push back on this please. I actually think having a single test with multiple combinations of `finalized` and `expired` is more useful than having individual test cases.  It also helps avoids unnecessary LOC.